src/bin/inx-chronicle/api/stardust/poi/merkle_hasher.rs

```diff
@@ -1,54 +1,133 @@
 // Copyright 2022 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
-use std::marker::PhantomData;
+use chronicle::types::stardust::block::BlockId;
+use crypto::hashes::{blake2b::Blake2b256, Digest, Output};
 
-use crypto::hashes::{Digest, Output};
+use super::{
+    merkle_proof::{Hashable, MerkleProof},
+    CreateProofError,
+};
 
 const LEAF_HASH_PREFIX: u8 = 0;
 const NODE_HASH_PREFIX: u8 = 1;
 
-/// A Merkle tree hasher that is generic over the hash function `H` being used.
-pub struct MerkleHasher<H> {
-    _phantom: PhantomData<H>,
-}
-
-impl<H: Default + Digest> MerkleHasher<H> {
-    pub fn new() -> Self {
-        Self { _phantom: PhantomData }
-    }
+/// A Merkle tree hasher.
+pub struct MerkleHasher;
```

Comment:
I think I'll take this change (otherwise I have a hard time arguing why the hasher should remain generic :sweat_smile: ).